test/bitswap.test.js

```diff
@@ -81,27 +193,100 @@ test('helia bitswap', async t => {
     }
   })
 
-  console.log('Creating local helia')
+  console.warn('Creating local helia')
   const helia = await createHelia({
     libp2p
   })
   const heliaFs = unixfs(helia)
+  return {
+    expected,
+    helia,
+    heliaFs,
+    hoverboard,
+    libp2p,
+    root
+  }
+}
 
-  console.log(`Dialing ${hoverboard}`)
-  const peer = await libp2p.dial(hoverboard)
-  t.is(peer.remoteAddr.getPeerId().toString(), peerId.id)
-
-  const decoder = new TextDecoder('utf8')
-  let text = ''
+/**
+ * test scenario that setups up a hoverboard running in a worker,
+ * a libp2p that can connect to it over websocket,
+ * and some example data to test with
+ * @param {object} options
+ * @param {object} [options.contentClaims] - content claims service
+ * @param {string} [options.contentClaims.url] - url of content claims service
+ */
+async function createHeliaBitswapScenarioMiniflare (
+  { contentClaims } = {},
+  { logLevel = LogLevel.DEBUG } = {}
+) {
+  const [dynamo, s3] = await Promise.all([createDynamo(), createS3()])
+  const [table, bucket] = await Promise.all([createDynamoTable(dynamo.client), createS3Bucket(s3.client)])
+  const builder = new Builder(dynamo.client, table, s3.client, 'us-west-2', bucket)
+  const encoder = new TextEncoder()
+  const expected = 'hoverboard 🛹'
+  const blob = new Blob([encoder.encode(expected)])
+  const root = await builder.add(blob)
 
-  console.log(`Fetching ${root}`)
-  for await (const chunk of heliaFs.cat(root)) {
-    text += decoder.decode(chunk, { stream: true })
+  const miniflareHttp = {
+    host: '127.0.0.1',
+    port: 63000 + Math.floor(Math.random() * 1000)
   }
+  const miniflare = new Miniflare({
```

Comment:
I don't think the workers-sdk `testWorker` fn that is unstable has a way of accessing the r2 bucket from outside the worker so I had to pul this in, and iiuc this new Miniflare 3 can only run on the built script in dist.